Framework/MDAlgorithms/test/ConvertMDHistoToMatrixWorkspaceTest.h

```diff
@@ -494,4 +494,101 @@ class ConvertMDHistoToMatrixWorkspaceTest : public CxxTest::TestSuite {
   }
 };
 
+/* Disabling clang format because the CXX parsser cannot read across multiple
+lines.
+ Clang formatting has therefore been disabled to prevent compiler errors.
+ When the parsser has been fixed (can read across multiple lines) the clang
+formatting
+can be enabled */
+
+// clang-format off
+class ConvertMDHistoToMatrixWorkspaceTestPerformance:public CxxTest::TestSuite {
+public:
+  static ConvertMDHistoToMatrixWorkspaceTestPerformance *createSuite() {
+    return new ConvertMDHistoToMatrixWorkspaceTestPerformance;
+  }
+  static void destroySuite(ConvertMDHistoToMatrixWorkspaceTestPerformance *suite) {
+    delete suite;
+  }
+//clang-format on
+  ConvertMDHistoToMatrixWorkspaceTestPerformance() {
+    std::vector<size_t> nonIntegr(2);
+    nonIntegr[0] = 0;
+    nonIntegr[1] = 1;
+    size_t ndims = 4;
+    size_t size = 1;
+
+    // property values for CreateMDHistoWorkspace
+    std::vector<size_t> numberOfBins(ndims);
+    std::vector<std::string> names(ndims);
+    // property values for SliceMDHisto
+    std::vector<coord_t> start(ndims);
+    std::vector<coord_t> end(ndims);
+    for (size_t i = 0; i < ndims; ++i) {
+      names[i] = "x_" + boost::lexical_cast<std::string>(i);
+      if (nonIntegr.end() != std::find(nonIntegr.begin(), nonIntegr.end(), i)) {
+        size_t nbins = 3 + i;
+        size *= nbins;
+        numberOfBins[i] = nbins;
+        // if it's a non-integrated dimension - don't slice
+        end[i] = static_cast<coord_t>(nbins);
+      } else {
+        numberOfBins[i] = 1;
+      }
+    }
+    signal_t signal(0.f), error(0.f);
+    // IMDHistoWorkspace_sptr slice =
+    slice = MDEventsTestHelper::makeFakeMDHistoWorkspaceGeneral(
+        ndims, signal, error, &numberOfBins.front(), &start.front(),
+        &end.front(), names);
+  }
+
+  void test_ConvertMDhistoToMatrixWorkspace() {
+
+    std::vector<size_t> nonIntegr(2);
+    auto alg =
+        AlgorithmManager::Instance().create("ConvertMDHistoToMatrixWorkspace");
+    alg->initialize();
+    alg->setRethrows(true);
+    alg->setChild(true);
+    alg->setProperty("InputWorkspace", slice);
+    alg->setPropertyValue("OutputWorkspace",
+                          "_2"); // Not really required for child algorithm
+
+    if (nonIntegr.size() > 2 || nonIntegr.empty()) {
+      TS_ASSERT_THROWS(alg->execute(), std::invalid_argument);
+    } else {
+      try {
+        alg->execute();
+      } catch (std::exception &e) {
+        TS_FAIL(e.what());
+      }
+
+      MatrixWorkspace_sptr matrix = alg->getProperty("OutputWorkspace");
+      TS_ASSERT(matrix);
+
+      if (nonIntegr.size() == 1) {
+        TS_ASSERT_EQUALS(matrix->getNumberHistograms(), 1);
```

Comment:
I wouldn't worry about performing these assertions here or below - if they are in the unit test we don't need them duplicated for the performance test.